pyinfra/operations/server.py

```diff
@@ -770,7 +770,7 @@ def group(group, present=True, system=False, gid=None, state=None, host=None):
 
         # Groups are often added by other operations (package installs), so check
         # for the group at runtime before adding.
-        yield "grep '{0}:' /etc/group || groupadd {1}".format(
+        yield "grep -E '^{0}:' /etc/group || groupadd {1}".format(
```

Comment:
```suggestion
        yield "grep '^{0}:' /etc/group || groupadd {1}".format(
```

I believe this should work OK, as the start/end characters work with non-extended-regex grep.

Reply:
Agreed. Done